src/view/treeNodes/fileChangeNode.ts

```diff
@@ -161,6 +163,19 @@ export class FileChangeNode extends TreeNode implements vscode.TreeItem {
 		this.changeModel.updateViewed(viewed);
 		this.contextValue = `${Schemes.FileChange}:${GitChangeType[this.changeModel.status]}:${viewed === ViewedState.VIEWED ? 'viewed' : 'unviewed'
 			}`;
+		this.checkboxState = viewed === ViewedState.VIEWED ?
+			vscode.TreeItemCheckboxState.Checked : vscode.TreeItemCheckboxState.Unchecked;
+	}
+
+	updateCheckbox(newState: vscode.TreeItemCheckboxState) {
+		const viewed = newState === vscode.TreeItemCheckboxState.Checked ? ViewedState.VIEWED : ViewedState.UNVIEWED;
+		this.updateViewed(viewed);
+		vscode.commands.executeCommand(
```

Comment:
Executing our own commands can indicate that we're circumventing some intentional layering and makes debugging and code exploration more difficult (no find references). In this case we can avoid the `executeCommand` by moving part of the commands into `FileChangeNode` as a public methods: 

https://github.com/microsoft/vscode-pull-request-github/blob/4204dd158ad2267548d7cb63db81ff9ddd4fc8fb/src/commands.ts#L908-L912